sig/steep/diagnostic/ruby.rbs

```diff
@@ -384,11 +384,11 @@ module Steep
       class IncompatibleAnnotation < Base
         attr_reader result: Subtyping::Result::Base
 
-        attr_reader relation: Subtyping::Relation
+        attr_reader relation: Subtyping::Relation[untyped]
```

Comment:
Maybe this can be typed stricter? Tell me if so.

Reply:
It may be... I'm not sure what the type should be. 🥲